File: BaseLibrary/command/equipment/GetRepairToolsAnswerCommand.cs
using System.Collections.Generic;
using BaseLibrary.equip;

namespace BaseLibrary.command.equipment {
    public class GetRepairToolsAnswerCommand : AEquipmentCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        /// <summary>
        /// Available repair tool to buy.
        /// </summary>
        /// <seealso cref="RepairTool"/>
        public RepairTool[] REPAIR_TOOLS { get; private set; }

        public GetRepairToolsAnswerCommand(RepairTool[] repairTools)
            : base() {
            REPAIR_TOOLS = repairTools;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/equipment/GetRepairToolsAnswerCommand.java

package BaseLibrary.communication.command.equipment;

import BaseLibrary.communication.protocol.AProtocol;
import BaseLibrary.equipment.RepairTool;

public class GetRepairToolsAnswerCommand extends AEquipmentCommand {

	private static final String NAME = "REPAIR_TOOLS_ANSWER";

	static {
		AProtocol.registerForDeserialize(NAME, GetRepairToolsAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private RepairTool[] repairTools;

	public GetRepairToolsAnswerCommand(RepairTool[] repairTools) {
		setRepairTools(repairTools);
	}
	
	/**
	 * Available repair tool to buy.
	 * 
	 * @see RepairTool
	 */
	public RepairTool[] getRepairTools() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return repairTools;
	}

	public void setRepairTools(RepairTool[] repairTools) {
		pending = false;
		this.repairTools = repairTools;
	}
}
